modules/exploits/multi/http/joomla_http_header_rce.rb

```diff
@@ -0,0 +1,177 @@
+    version = Gem::Version.new(php_version)
+    vulnerable = false
+
+    # check for ubuntu and debian specific versions. Was fixed in
+    # * 5.5.9+dfsg-1ubuntu4.13
+    # * 5.3.10-1ubuntu3.20
+    # * 5.4.45-0+deb7u1
+    # Changelogs (search for CVE-2015-6835 or #70219):
+    #   http://changelogs.ubuntu.com/changelogs/pool/main/p/php5/php5_5.5.9+dfsg-1ubuntu4.13/changelog
+    #   http://changelogs.ubuntu.com/changelogs/pool/main/p/php5/php5_5.3.10-1ubuntu3.20/changelog
+    #   http://metadata.ftp-master.debian.org/changelogs/main/p/php5/php5_5.4.45-0+deb7u2_changelog
+    if rest && rest.include?('ubuntu')
+      sub_version = rest.scan(/^\dubuntu([\d\.]+)/i).flatten.first || ''
+      vprint_status("Found Ubuntu PHP version: #{res.headers['X-Powered-By']}")
+
+      if version > Gem::Version.new('5.5.9')
+        vulnerable = false
+      elsif version == Gem::Version.new('5.5.9') && Gem::Version.new(sub_version) >= Gem::Version.new('4.13')
+        vulnerable = false
+      elsif version == Gem::Version.new('5.3.10') && Gem::Version.new(sub_version) >= Gem::Version.new('3.20')
+        vulnerable = false
+      else
+        vulnerable = true
+      end
+    elsif rest && rest.include?('+deb')
+      sub_version = rest.scan(/^\d+\+deb([\du]+)/i).flatten.first || ''
+      vprint_status("Found Debian PHP version: #{res.headers['X-Powered-By']}")
+
+      if version > Gem::Version.new('5.4.45')
+        vulnerable = false
+      elsif version == Gem::Version.new('5.4.45') && sub_version != '7u1'
+        vulnerable = false
+      else
+        vulnerable = true
+      end
+    else
+      vprint_status("Found PHP version: #{res.headers['X-Powered-By']}")
+      vulnerable = true if version <= Gem::Version.new('5.4.44')
+      vulnerable = true if version.between?(Gem::Version.new('5.5.0'), Gem::Version.new('5.5.28'))
+      vulnerable = true if version.between?(Gem::Version.new('5.6.0'), Gem::Version.new('5.6.12'))
+    end
+
+    unless vulnerable
+      vprint_error('This module currently does not work against this PHP version')
+      return Exploit::CheckCode::Safe
+    end
+
+    res = send_request_cgi({'uri' => normalize_uri(target_uri.path, 'administrator', 'manifests', 'files', 'joomla.xml') })
+    if res && res.code == 200 && res.body && res.body.include?('<author>Joomla! Project</author>')
+      joomla_version = res.body.scan(/<version>([\d\.]+)<\/version>/i).flatten.first || ''
+      if joomla_version
+        return Exploit::CheckCode::Vulnerable if Gem::Version.new(joomla_version) < Gem::Version.new('3.4.6')
+      end
+    end
+
+    res.get_html_meta_elements.each do |element|
+      if element.attributes['name'] &&
+        /^generator$/i === element.attributes['name'] &&
+        element.attributes['content'] &&
+        /joomla/i === element.attributes['content'].value
+        return Exploit::CheckCode::Detected
+      end
+    end
+
+    Exploit::CheckCode::Unknown
```

Comment:
Exploit::CheckCode::Unknown is sort of meant to be used when you're in the middle of retrieving some info, but fail in the process (like a timeout, connection problem, or unexpected response). So I would suggest Exploit::CheckCode::Safe again... unless there is a special reason for it.

Please see:
https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check%28%29-method